pkg/pillar/docs/nim.md

```diff
@@ -0,0 +1,192 @@
+# Network Interface Manager (NIM)
+
+## Overview
+
+NIM manages (physical) network interfaces (Ethernet, WiFi, LTE) of a device based
+on configuration coming from various sources (controller, override, last-resort,
+persisted config).
+Its mail goal is to preserve connectivity between the device and the controller.
+NIM verifies new configuration changes before fully committing to them.
+Moreover, it maintains one or more old but working configurations with a lower-priority,
+allowing to fallback in case the latest config breaks connectivity, but always tries
+to move to the most recent, aka the highest-priority configuration.
+
+## Key Input/Output
+
+**NIM consumes**:
+
+* configuration for device ports
+  * an instance of the `DevicePortConfig` struct (abbreviated to DPC)
+  * contains configuration for every physical interface, except those which are
+    disabled or directly assigned to applications
+  * DPC is received from different sources, such as zedagent (config from controller),
+    the `/config` partition with override DPC copied from USB stick
+    by `device-steps.sh` and even from NIM itself, which builds and publishes
+    the *last-resort* config if enabled
+* global configuration properties
+  * an instance of `ConfigItemValueMap` struct received from zedagent
+  * used to determine if last-resort should be enabled, also to obtain time
+    intervals/timeouts to apply for DPC verification and finally to get parameters
+    for SSH access
+* a list of assignable adapters
+  * an instance of `AssignableAdapters` struct received from domainmgr
+  * used to determine if a given interface is assigned to pciback or if it is available
+    to the host (EVE) OS
+* zedagent status
+  * an instance of `ZedAgentStatus` received from zedagent
+  * used to determine the intended state of the [Radio-Silence mode](./radio-silence.md)
+
+**NIM publishes**:
+
+* the current state of the device connectivity
+  * an instance of the `DeviceNetworkStatus` struct (abbreviated to DNS)
+  * published whenever the config or the state of connectivity/interfaces changes
+* the currently maintained list of Device Port Configurations (DPCs)
```

Comment:
Makes sense to state that this is a persistent publication (so that NIM can read it back in after a reboot).

Reply:
done